File: backend/server-py/app/src/db/schemas/gyoan_words_map.py
from .base import Base, TimestampMixin
import sqlalchemy as sa

class GyoanWordsMap(Base, TimestampMixin):
    __tablename__ = 'gyoan_words_map'
    id = sa.Column(sa.Integer, primary_key=True)
    gyoan_id = sa.Column(sa.Integer, sa.ForeignKey('gyoan.id'), nullable=False)
    words_id = sa.Column(sa.Integer, sa.ForeignKey('words.id'), nullable=False)

File: backend/server-node/src/db/schemas/gyoan_words_map.js
const { EntitySchema } = require('typeorm');

const GyoanWordsMapSchema = new EntitySchema({
    name: 'GyoanWordsMap',
    tableName: 'gyoan_words_map',
    columns: {
        id: {
            primary: true,
            type: 'int',
            generated: true
        },
        gyoan_id: {
            type: 'int',
            nullable: false
        },
        words_id: {
            type: 'int',
            nullable: false
        },
        created_at: {
            type: 'timestamp',
            createDate: true,
            nullable: false
        },
        updated_at: {
            type: 'timestamp',
            updateDate: true,
            nullable: false
        }
    },
    relations: {
        gyoan: {
            target: 'Gyoan',
            type: 'many-to-one',
            joinColumn: {
                name: 'gyoan_id',
                referencedColumnName: 'id'
            },
            nullable: false
        },
        words: {
            target: 'Words',
            type: 'many-to-one',
            joinColumn: {
                name: 'words_id',
                referencedColumnName: 'id'
            },
            nullable: false
        }
    }
});

module.exports = GyoanWordsMapSchema;
